tests/picard_test.py

```diff
@@ -0,0 +1,220 @@
+    tol = 1e-6
+    sol, state = PicardIteration(f, maxiter=100, tol=tol).run(x0)
+    self.assertLess(state.error, tol)
+    sol_norm = tree_l2_norm(sol)
+    self.assertLess(sol_norm, tol)
+    f_fixed_point = jnp.zeros(3), jnp.zeros((2,1))
+    self.assertAllClose(sol, f_fixed_point)
+    #scipy_f = lambda x: jnp.sin(x) * x**2
+    #self._compare_scipy(sol[0], scipy_f, x0[0], history_size=3, maxiter=None, x_tol=1e-5, atol=1e-4)
+
+  def test_cos_fixed_point(self):
+    """Test convergence for cos fixed point (non-zero fixed point).
+    
+    Also test support for additional parameters.
+    """
+    def f(x, theta):
+      return jnp.cos(x) + theta
+    x0 = jnp.array([0.3, jnp.pi / 4])
+    theta = jnp.array([0., jnp.pi / 2])
+    f_fixed_point = jnp.array([0.73908614842288, jnp.pi / 2])
+    tol = 1e-2
+    sol, state = PicardIteration(f, maxiter=100000, tol=tol).run(x0, theta)
+    self.assertLess(state.error, tol)
+    self.assertArraysAllClose(sol, f_fixed_point, atol=1e-2)
+    #scipy_f = lambda x: f(x, theta)
+    #self._compare_scipy(sol, scipy_f, x0, history_size=4, maxiter=None, x_tol=1e-6, atol=1e-2)
+
+  def test_gradient_descent(self):
+    """Fixed point of map T(x)=x-grad_f(x) corresponding to gradient step.
+    
+    f is chosen to be some least square regression on dummy dataset.
+    """
+    X, y = datasets.make_regression(n_samples=10, n_features=3, random_state=0)
+    least_square_grad = jax.grad(objective.least_squares)
+    def T_map(param, eta, *args, **kwargs):
+      g = least_square_grad(param, *args, **kwargs)
+      step = tree_scalar_mul(eta, g)
+      return tree_sub(param, step)
+    w_init = jnp.zeros(X.shape[1])
+    eta = 1e-1  # reasonable step size
+    tol = 1e-5
+    aa = PicardIteration(T_map, maxiter=300, tol=tol)
+    sol, state = aa.run(w_init, eta, (X, y))
+    self.assertLess(state.error, tol)
+    grad = least_square_grad(sol, (X,y))
+    zero = jnp.zeros_like(w_init)
+    self.assertArraysAllClose(grad, zero, atol=1e-4)
+    #scipy_f = lambda param: T_map(param, eta, (X, y))
+    #self._compare_scipy(sol, scipy_f, w_init, history_size=5, maxiter=None, x_tol=1e-4, atol=1e-4)
+
+  def test_filter_leaves(self):
+    """Test support of ``filter_leaves`` functionnality."""
+    def f(params, r):
+      x, y = params
+      fx = jnp.cos(x)  # fixed point 0.739085
+      fy = r * y * (1 - y)  # logistic map: chaotic behavior
+      return fx, fy
+    r = jnp.array([3.95])  # for this value there is chaotic behavior
+    x0logistic = jnp.array([1.]), jnp.array([0.6])
+    tol = 2e-2
+    aa = PicardIteration(f, maxiter=400*1000, tol=tol, filter_leaves=(True,False))
+    sol, state = aa.init(x0logistic, r)
+    sols = []
+    for i in range(10):
+      sol, state = aa.update(sol, state, r)
+      sols.append(sol)
+    self.assertLess(state.error, tol)
+    self.assertArraysAllClose(sol[0], jnp.array([0.739085]), atol=1e-2)
+
+  def test_anderson_ode(self):
```

Comment:
Does this test anything that is not covered by other tests? If not not I would remove it. Otherwise, anderson -> picard.